packages/pyramid_lint/lib/src/lints/flutter/dispose_controllers.dart

```diff
@@ -224,20 +224,33 @@ Iterable<String> _getDisposeStatementTargetNames(
   NodeList<Statement> statements,
 ) {
   return statements.expressionStatements
-      .map((e) => e.expression)
-      .whereType<MethodInvocation>()
       .map(_getTargetNameOfDisposeMethodInvocation)
       .nonNulls;
 }
 
 String? _getTargetNameOfDisposeMethodInvocation(
-  MethodInvocation invocation,
+  ExpressionStatement expressionStatement,
 ) {
-  final target = invocation.target;
-  if (target is! SimpleIdentifier) return null;
+  if (expressionStatement.expression is CascadeExpression) {
+    final cascadeExpression =
+        expressionStatement.expression as CascadeExpression;
 
-  final methodName = invocation.methodName.name;
-  if (methodName != 'dispose') return null;
+    for (final section in cascadeExpression.cascadeSections) {
+      if (section is MethodInvocation && section.methodName.name == 'dispose') {
+        return cascadeExpression.target.toString();
```

Comment:
We probably don't want to use `toString` here to get the target name.
Instead we could first check if the target of the `CascadeExpression` is a `SimpleIdentifier` or not and use `SimpleIdentifier.name` to get the name.

Reply:
Let's do that @charlescyt, thanks for the suggestion!